### PR TITLE
Fixing the docs

### PR DESCRIPTION
**Pull request by Q-lds**
This PR:
- Fixes many bugs in the docs (especially in the utilities)
- Fixes the math display in the 4th notebook
- Adds the flag -W to the doc tests, which should fail if any warning is raised

This PR addresses
#111 #106

### CODE REVIEW

**Comment by Q-lds** [comment]
@vishal-ph, the PR is ready for review!

File: docs/notebook_converter.sh [hunk 1]
```diff
@@ -1,9 +0,0 @@
-
```
Comment (vishal-ph) [inline docs/notebook_converter.sh:1]: Why is this file removed? Was this file generating the notebooks for `readthedocs`?

Comment (Q-lds) [reply docs/notebook_converter.sh:1]: It wasn't actually used. The list of commands executed by readthedocs (see https://github.com/entropicalabs/openqaoa/blob/feat_fix_docs/.readthedocs.yml#L17) and the action that tests the docs is already doing it anyway (see https://github.com/entropicalabs/openqaoa/blob/feat_fix_docs/.github/workflows/test_dev.yml#L108)

File: openqaoa/qaoa_parameters/standardparams.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ class QAOAVariationalStandardParams(QAOAVariationalBaseParams):
     hyperparameters:
         The hyperparameters containing the register, terms, weights, and the number of layers
         ``hyperparameters = (register, terms, weights, p)``
-    parameters:
+    qaoa_circuit_params:
```
Comment (vishal-ph) [inline openqaoa/qaoa_parameters/standardparams.py:42]: okay, I see this is incorrect. This is probably outdated docstring from the outdated script. Let's fix this

Comment (Q-lds) [reply openqaoa/qaoa_parameters/standardparams.py:42]: Yes, of course! Feel free to edit and push

**Review by vishal-ph — APPROVED** [review]
